### PR TITLE
Vulnerability check all packages and dependencies served by dapla-pypiserver

### PR DESCRIPTION
**Pull request by drageset**
TL;DR: 

We (@mmj-ssb and I) have implemented vulnerability scanning of the python packages that are served by dapla-pypiserver. The scan happens in the docker build process, and fails the build if vulnerabilities are found.

Long version:

A python script compiles a standardized python `requirements.txt` of all the actual versions of the packages (incl. dependencies) that are downloaded to the pypiserver based on the whitelist `Packages.txt`. This generated `requirements.txt` is run through a regular `safety check -r requirements.txt`, which will fail the Docker build process if there are vulnerable package versions listed. The script to compile the requirements took about 5 seconds to complete for the current whitelist on our personal development machines. There are 87 lines in the whitelist, and 381 dependencies in the requirements.txt.

!NB!

At the moment, `keras==2.4.3` which is explicitly listed in the whitelist is vulnerable, and thus fails the build. If we are to take this seriously, this means that we will remove `keras==2.4.3` from the whitelist despite the fact that some user might depend on it. Will we?

---------

Misadventures: 

On this branch we have tried two other alternate solutions to scanning. The solution where each package is installed and scanned in a virtual environment took 75 minutes to complete, and did not work due to issues(possible bugs) with `pipenv check` (python safety check functionality for a pipenv virtual environment), which we remediated by using jq to extract the package name and version number from the `Pipfile.lock`, and compiling them into a master "requirements.txt" that we scanned with regular `safety check -r requirements.txt`.  The spirit of this approach was then adopted into the python script that used `pkginfo` to extract that same information without installing or unpacking the python wheel/source distribution. The reason we could not use the package name and version from the filename of the distribution is because those names are not guaranteed to be uniform/correct. The other solution we tried using `depshell` was complex, not necessarily reliable, and based on an archived/unmaintained project. The code relating to the alternate solutions was deleted from the branch before the pull request was submitted.

### CODE REVIEW

**Review by arneso-ssb — APPROVED** [review]
Looks good! Good work! :-)

**Review by bjornandre — APPROVED** [review]
Looks good to me! Nice work and documentation 👍
